internal/core/cloud/azure.go

```diff
@@ -220,7 +221,11 @@ func NewAzureMetadata(client HTTPClient) (*AzureMetadata, error) {
 		},
 	}
 
-	req, _ := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s/metadata/instance", azureAPIAddress), nil)
+	req, _ := http.NewRequestWithContext(
+		ctx, http.MethodGet,
```

Comment:
I would move `http.MethodGet` to a new line

Reply:
done